pkg/arvo/sys/vane/eyre.hoon

```diff
@@ -1910,8 +1910,29 @@
     [~ http-server-gate]
   ::  %trim: in response to memory pressure
   ::
+  ::    Cancel all inactive channels
+  ::    XX cancel active too if =(0 trim-priority) ?
+  ::
   ?:  ?=(%trim -.task)
-    [~ http-server-gate]
+    =/  event-args  [[our eny duct now scry-gate] server-state.ax]
+    =*  on-channel-timeout
+      on-channel-timeout:by-channel:(per-server-event event-args)
+    =*  channel-state  channel-state.server-state.ax
+    ::
+    =/  inactive=(list @t)
+      =/  full=(set @t)  ~(key by session.channel-state)
+      =/  live=(set @t)
+        (~(gas in *(set @t)) ~(val by duct-to-key.channel-state))
+      ~(tap in (~(dif in full) live))
+    ::
+    =|  moves=(list (list move))
+    |-  ^-  [(list move) _http-server-gate]
+    ?~  inactive
+      ~>  %slog.[0 leaf+"eyre: trim: closing inactive channels"]
+      [(zing (flop moves)) http-server-gate]
+    =^  mov  server-state.ax  (on-channel-timeout i.inactive)
```

Comment:
If you're raw calling on-channel-timeout, I don't think this clears the behn timer which is going to call it in the future.

Reply:
ah, thanks.

Reply:
Done, PTAL